src/wl-copy.c

```diff
@@ -249,45 +249,43 @@ void do_offer
 }
 
 void init_selection(char *mime_type) {
-    if (use_wlr_data_control) {
 #ifdef HAVE_WLR_DATA_CONTROL
-        struct zwlr_data_control_source_v1 *data_control_source =
-            zwlr_data_control_manager_v1_create_data_source(
-                data_control_manager
-            );
-        zwlr_data_control_source_v1_add_listener(
-            data_control_source,
-            &data_control_source_listener,
-            NULL
+    struct zwlr_data_control_source_v1 *data_control_source =
```

Comment:
No, we must check whether the compositor supports wlr-data-control at all :smile: